### DISCUSSION TITLE
Magnitude of the gradient of the unknown variable

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23693] [Q&A General] [answered]
Hello everyone,

I need to get the magnitude of the gradient of the unknown variable (abs(_grad_u)) to compute the residual term in the weak form of the equation. Is there any available object so I can get the abs(_grad_u) or I should do it myself?

Thanks,
Ali

**Comment by GiudGiud** [comment, marked as answer]
Hello

don’t think we have that one but you could save the gradient in an intermediate aux variable, then use another auxkernel to get the magnitude of the vector. 

It will only cost a little bit of extra memory to do that 

Guillaume
